cpp/src/gandiva/regex_functions_holder.cc

```diff
@@ -275,4 +275,78 @@ const char* ExtractHolder::operator()(ExecutionContext* ctx, const char* user_in
   return result_buffer;
 }
 
+Result<std::shared_ptr<RegexpLikeHolder>> RegexpLikeHolder::Make(
+    const FunctionNode& node) {
+  ARROW_RETURN_IF(
+      node.children().size() < 2,
+      Status::Invalid("'regexp_like' function requires at least two parameters"));
+  auto pattern = dynamic_cast<LiteralNode*>(node.children().at(1).get());
```

Comment:
Although it's not prevalent in Gandiva, but we typically prefer `arrow::internal::CheckedCast` for better performance